Improve installation and testing documentation in README

ayuugoyal opened the pull request:
## Overview
This PR enhances the README.md documentation for the aruco_ros repository by adding comprehensive installation and setup instructions while preserving all existing content. These improvements aim to lower the barrier to entry for new users and make the repository more accessible.

## Changes Made
- Added structured installation instructions covering:
  - Prerequisites
  - ROS installation reference (generalized for all distros)
  - Workspace creation
  - Repository cloning
  - Dependency installation
  - Build process for both ROS 1 and ROS 2
- Added instructions for installing REEM robot simulation from source repositories
- Added testing instructions for using aruco_ros with regular cameras
- Enhanced existing testing instructions with helpful inline comments
- Improved code block formatting with bash language specification
- Added troubleshooting section

## Benefits
- New users can quickly get started with the package
- Clearer instructions reduce potential installation issues
- Generic distro approach makes documentation future-proof
- Preserves all original documentation while enhancing usability
- Follows the repository's existing documentation style

## Testing Done
I've verified that these installation instructions work properly on Ubuntu 20.04 with ROS Noetic and Ubuntu 22.04 with ROS Humble. All commands and paths have been carefully checked for accuracy.

## Motivation
I'm interested in contributing to this project as part of my GSoC application. After reviewing the repository, I identified documentation as an area where I could make an immediate positive impact while familiarizing myself with the codebase.

I look forward to any feedback and am happy to make additional changes as needed.

ayuugoyal committed:
Improve installation and testing documentation in README

ayuugoyal commented:
@saikishor can you review this?

saikishor commented on README.md:
```diff
@@ -1,27 +1,106 @@
- * Visual servoing: track object and hand at the same time
+### Prerequisites
+- A compatible Ubuntu version for your chosen ROS distribution
+- ROS installed (ROS 1 or ROS 2)
```
Could you mention the OpenCV versions as well?

ayuugoyal replied:
sure will do

saikishor commented on README.md:
````diff
@@ -1,27 +1,106 @@
+mkdir -p ~/aruco_ws/src
+cd ~/aruco_ws/src
+```
+
+### 3. Clone the aruco_ros Repository
+
+```bash
+# Clone the repository
+git clone https://github.com/pal-robotics/aruco_ros.git -b <distro>-devel
+```
+Replace `<distro>` with your ROS distribution name (e.g., melodic, noetic, humble)
+
+### 4. Install Dependencies
+
+```bash
+# Navigate to the workspace root
+cd ~/aruco_ws
+# Install dependencies using rosdep
+rosdep install --from-paths src --ignore-src -r -y
+```
+
+### 5. Build the Workspace
+
+For ROS 1:
+```bash
+# Build the workspace using catkin_make
+cd ~/aruco_ws
+catkin_make
+# Source the setup file to update environment
+source devel/setup.bash
+```
+
+For ROS 2:
+```bash
+# Build the workspace using colcon
+cd ~/aruco_ws
+colcon build
+# Source the setup file to update environment
+source install/setup.bash
+```
````
All of this, I would set under a section Build from Source 

And it would be great, if you can add a section "Using Binaries"

ayuugoyal replied:
will do the required changes

saikishor commented on README.md:
````diff
@@ -1,27 +1,106 @@
+source install/setup.bash
+```
+
+### 6. Installing REEM Robot Simulation (Optional)
+
+If you want to test with the REEM robot as in the examples below, you'll need to install the REEM robot packages from source:
+
+```bash
+# Create a workspace for REEM if you don't want to use your existing one
+mkdir -p ~/reem_ws/src
+cd ~/reem_ws/src
+
+# Clone the necessary repositories
+git clone https://github.com/pal-robotics/reem_robot.git
+git clone https://github.com/pal-robotics/reem_simulation.git
+git clone https://github.com/pal-robotics/pal_gazebo_worlds.git -b <distro>-devel
+
+# Install dependencies
+cd ~/reem_ws
+rosdep install --from-paths src --ignore-src -r -y
+
+# Build the workspace
+# For ROS 1:
+catkin_make
+source devel/setup.bash
+
+# For ROS 2:
+colcon build
+source install/setup.bash
+```
+
+Replace `<distro>` with your ROS distribution name (e.g., noetic, humble)
````
Remove this part

ayuugoyal replied:
Thanks for reviewing this!

saikishor reviewed (commented):
Thank you so much for your contribution.
Very much appreciated 

It would be cool, if you can modify to have only for ROS 2 , as ROS 1 is about to EOL

ayuugoyal committed:
docs: Improve README documentation with installation instructions

ayuugoyal commented:
@saikishor  I've made all the requested changes to the README:

1. Removed ROS 1 references and noted that it's approaching EOL
2. Added OpenCV version information in the prerequisites section
3. Reorganized the installation instructions into "Using Binaries" and "Build from Source" sections
4. Removed the "Installing REEM Robot Simulation" section completely

Could you please verify these changes? Let me know if anything else needs adjustment.

ayuugoyal commented:
@saikishor Did you get a chance to review this? If everything looks good, could you please close the PR? Thanks!
